### PR TITLE
Add contributing.md

### PR DESCRIPTION
**Pull request by bdarcus**
I'm starting this, and posting this draft for now for people to think about.

How can we make our work easier, and the process more clear for new contributors?

My thought is we fill this out with details, and that we add templates for issue creation.

### CODE REVIEW

**Comment by bdarcus** [comment]
A bit unclear is how this relates to csl-evolution.

**Comment by bdarcus** [comment]
This is related to #220, as we need to include instructions here on what we conclude; about our guidelines for commits, and commit messages, on PRs.

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+- in general, please follow [these recommendations](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) on writing good commit messages
+- reference existing issues, or create and reference new ones, that clearly lay out the issue your PR is addressing
+- if you are merely adding a string to `csl-terms.rnc`, `csl-types.rnc`, or `csl-variables`:
+   - please start your commit message with "Add" and specify which type of string you are adding; for example `Add "foo" and "bar" type`
+   - if you find your first line commit message is over 50 characters, that may suggest you PR should be split up
+   - make sure to add the string(s) to the `csl-data.json` schema as well in your commit
+- if you are proposing any change to `csl.rnc`, be careful. This is the file that describes the basic model of CSL, and so changes here have the potential to break implementations. If you feel you must request a breaking change, please include the word "breaking" in the body of your commit message, as our changelog tool will pick that up.
+- if you are suggesting deprecating anything, please including the word "deprecate" in the body of your commit message
```
Comment (bdarcus) [inline contributing.md:27]: maybe add a "types" section here, with possible options of:

- add (new features for string)
- chore (for non core core code changes)
- deprecate
- ???

**Comment by denismaier** [comment]
Ahh. That was too quick. It is still a draft. Can I remove the approval?

**Comment by denismaier** [comment]
Oh... Merged already... fine

**Comment by bdarcus** [comment]
Yeah, but we can always revise going forward.

**Pull request merged**